### PR TITLE
mathemagix: init at 11126

### PR DESCRIPTION
**Pull request by drupol**
TODO:

- [x] ~~Currently broken because it tries to write in the user directory at the end of compilation... investigation ongoing.~~ Fixed
- [x] Add supported platforms (linux only)
- [x] Fix https://discourse.nixos.org/t/building-mathemagix/43846/4
- [ ] Licencing (I wrote the author on the 22 April in the morning... no news so far)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/mathemagix/package.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  ];
+
+  preConfigure = ''
+    export HOME="$PWD"
+  '';
+
+  configureFlags = [ "--prefix=${placeholder "out"}" ];
+
+  meta = {
+    description = "Mathemagix is a free computer algebra and analysis system. It consists of a high level language with a compiler and a series of mathematical libraries, some of which are written in C++.";
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/mathemagix/package.nix:39]:
```suggestion
    description = "A free computer algebra and analysis system consisting of a high level language with a compiler and a series of mathematical libraries";
```

https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

File: pkgs/by-name/ma/mathemagix/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  pname = "mathemagix";
+  version = "11126";
+
+  src = fetchsvn {
+    url = "https://subversion.renater.fr/anonscm/svn/mmx/";
+    rev = 11126;
+    hash = "sha256-AFnYd5oFg/wgaHPjfZmqXNljEpoFW4h6f3UG+KZauEs=";
+  };
+
+  nativeBuildInputs = [
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/mathemagix/package.nix:23]: Some of these should be put inside `buildInputs`? Please enable `strictDeps` and try again.

File: pkgs/by-name/ma/mathemagix/package.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    libtool
+    gmp
+    mpfr
+  ];
+
+  preConfigure = ''
+    export HOME="$PWD"
+  '';
+
+  configureFlags = [ "--prefix=${placeholder "out"}" ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/mathemagix/package.nix:36]: Already the default? See https://nixos.org/manual/nixpkgs/unstable/#variables-controlling-the-configure-phase

File: pkgs/by-name/ma/mathemagix/package.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  meta = {
+    description = "Mathemagix is a free computer algebra and analysis system. It consists of a high level language with a compiler and a series of mathematical libraries, some of which are written in C++.";
+    homepage = "http://www.mathemagix.org/";
+    license = lib.licenses.gpl3Only;
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/mathemagix/package.nix:41]: Any source? https://subversion.renater.fr/anonscm/svn/mmx/numerix/install-sh and https://subversion.renater.fr/anonscm/svn/mmx/configure for example are not GPL3

File: pkgs/by-name/ma/mathemagix/package.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  mpfr,
+}:
+
+stdenv.mkDerivation {
+  pname = "mathemagix";
+  version = "11126";
+
+  src = fetchsvn {
+    url = "https://subversion.renater.fr/anonscm/svn/mmx/";
+    rev = 11126;
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/mathemagix/package.nix:19]:
```suggestion
    rev = lib.toInt finalAttrs.version;
```

(not necessary, needing finalAttrs)

File: pkgs/by-name/ma/mathemagix/package.nix (hunk 4 above)
Comment (drupol) [reply pkgs/by-name/ma/mathemagix/package.nix:44]: As far I've seen, all the licenses used in the project are GPL (https://subversion.renater.fr/anonscm/svn/mmx/mmxlight/LICENSE, https://subversion.renater.fr/anonscm/svn/mmx/jorix/LICENSE, https://subversion.renater.fr/anonscm/svn/mmx/caas/LICENSE, etc etc... ).

**Comment by adam-neeley** [comment]
Nice job drupol and friends. This really exceeded my expectations. I'll be keeping an eye on this.

File: pkgs/by-name/ma/mathemagix/package.nix (hunk 4 above)
Comment (drupol) [reply pkgs/by-name/ma/mathemagix/package.nix:44]: Are we all OK to move on as it is now and refine later on if needed?

**Comment by drupol** [comment]
Are we all OK to move on and merge this PR and eventually refine it later if needed?

File: pkgs/by-name/ma/mathemagix/package.nix (hunk 4 above)
Comment (adam-neeley) [reply pkgs/by-name/ma/mathemagix/package.nix:44]: Yes, all LICENSE files I've seen specify GPL3.

The LICENSE file in the directory of /mmx/numerix/install-sh is GPL3, so I believe install-sh is in fact GPL3.

While /mmx/configure may not directly specify GPL3, it does state
```
# This configure script is free software; the Free Software Foundation
# gives unlimited permission to copy, distribute and modify it.
```

Clearly the author intended for this software to be freely distributed. Getting ahold of academic researchers is often a difficult task. If people are uneasy about a configuration script or two not being marked as GPL3, perhaps there is a different licensing option we can use.

Personally I think this is a non-issue. I think this package is ready to be merged.

**Pull request merged**